src/routes/webhooks/customerio.ts

```diff
@@ -153,6 +164,42 @@ export const customerio = async (fastify: FastifyInstance): Promise<void> => {
     { prefix: '/marketing_cta' },
   );
 
+  fastify.post<PromotedPostPayload>('/promote', {
+    config: {
+      rawBody: true,
+    },
+    handler: async (req, res) => {
+      const valid = verifyCIOSignature(
+        process.env.CIO_REPORTING_WEBHOOK_SECRET,
+        req,
+      );
+      if (!valid) {
+        req.log.warn('cio reporting webhook invalid signature');
+        return res.status(403).send({ error: 'Invalid signature' });
+      }
+
+      try {
+        const { userId, postId } = req.body;
+
+        const expirationPeriod = ONE_WEEK_IN_SECONDS * 1000; // maybe need to make it configurable
+        const currentDate = new Date();
+        const validUntil = new Date(currentDate.getTime() + expirationPeriod);
```

Comment:
```suggestion
        const validUntil = addDays(new Date, 7);
```

This could also work. `addDays` is imported from `date-fns`